presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/TypeValidator.java

```diff
@@ -140,11 +147,33 @@ public Void visitUnion(UnionNode node, Void context)
             return null;
         }
 
-        private void checkWindowFunctions(Map<Symbol, WindowNode.Function> functions)
+        private void checkWindowFunctions(Map<Symbol, WindowNode.Function> functions, WindowNode.Specification specification)
         {
+            // Inheritance of the following WindowSpecification components is done by the analyzer. As this information
```

Comment:
Similarly, putting additional tests in TestMergeAdjacentWindows and TestSwapAdjacentWindowsBySpecifications for the inheriting case seems like a good plan.